### ISSUE TITLE
`flet build` gives the following error.

### ISSUE THREAD

**Issue by frowNYfaceUnknown** [flet-dev/flet#2309]
**Description**
My flet app tests as expected, using `flet run [--ios] [--android]`, on all the following platforms:

- windows,
- web,
- ios and
- andoid.

Now, when I am trying to build the application through the `flet build apk` or `flet build windows` command it throws me an error.

**Describe the results you received:**
The error I get using `flet build windows`:

```
Creating Flutter bootstrap project...OK
Customizing app icons and splash images...OK
Generating app icons...OK
Packaging Python app...Running package command
Creating asset directory: C:\Users\aarya\AppData\Local\Temp\flet_flutter_build_DPfhwLfHuo\app
Copying Python app from E:\flet_venv\jinkushal to C:\Users\aarya\AppData\Local\Temp\serious_python_temp3df25986
Installing dependencies  with pip command to
C:\Users\aarya\AppData\Local\Temp\serious_python_temp3df25986\__pypackages__
Extracting Python distributive from
C:\Users\aarya\AppData\Local\Temp\cpython-3.11.6+20231002-x86_64-pc-windows-msvc-shared-install_only.tar.gz to
C:\Users\aarya\AppData\Local\Temp\hostpython3.11_fd297015
Compiling Python sources at C:\Users\aarya\AppData\Local\Temp\serious_python_temp3df25986

Error building Flet app - see the log of failed command above.
```

The error I get using `flet build windows -vv`

```
Flutter bootstrap directory: C:\Users\aarya\AppData\Local\Temp\flet_flutter_build_OYzPOnxyAd
Creating Flutter bootstrap project...OK
Customizing app icons and splash images...OK
Generating app icons...Resolving dependencies in C:\Users\aarya\AppData\Local\Temp\flet_flutter_build_OYzPOnxyAd... (1.0(6.1s)
Got dependencies in C:\Users\aarya\AppData\Local\Temp\flet_flutter_build_OYzPOnxyAd.
Building package executable... (6.4s)
Built flutter_launcher_icons:flutter_launcher_icons.
 ════════════════════════════════════════════
FLUTTER LAUNCHER ICONS (v0.13.1)
════════════════════════════════════════════
• Creating default icons Android
• Overwriting the default Android launcher icon with a new icon
• Overwriting default iOS launcher icon with new icon
Creating Icons for Web...
Creating Icons for Windows...
Creating Icons for MacOS...
✓ Successfully generated launcher icons
OK
Packaging Python app...Building package executable... (4.1s)
Built serious_python:main.
Running package command
Creating asset directory: C:\Users\aarya\AppData\Local\Temp\flet_flutter_build_OYzPOnxyAd\app
Copying Python app from E:\flet_venv\jinkushal to C:\Users\aarya\AppData\Local\Temp\serious_python_tempb43f0909
Installing dependencies [aenum==3.1.15, anyio==4.2.0, arrow==1.3.0, binaryornot==0.4.4, certifi==2023.7.22, chardet==5.2.0, charset-normalizer==3.3.2, click==8.1.7, colorama==0.4.6, cookiecutter==2.5.0, dnspython==2.4.2, h11==0.14.0, httpcore==0.17.3, httpx==0.24.1, idna==3.4, Jinja2==3.1.2, markdown-it-py==3.0.0, MarkupSafe==2.1.3, mdurl==0.1.2, oauthlib==3.2.2, packaging==23.1, Pygments==2.17.2, pymongo==4.6.1, pypng==0.20220715.0, python-dateutil==2.8.2, python-slugify==8.0.1, PyYAML==6.0.1, qrcode==7.4.2, repath==0.9.0, requests==2.31.0, rich==13.7.0, six==1.16.0, sniffio==1.3.0, text-unidecode==1.3, types-python-dateutil==2.8.19.14, typing_extensions==4.7.1, urllib3==2.1.0, watchdog==3.0.0, websocket-client==1.7.0, websockets==11.0.3, pytz==2023.3, flet-embed] with pip command to
C:\Users\aarya\AppData\Local\Temp\serious_python_tempb43f0909\__pypackages__
Extracting Python distributive from C:\Users\aarya\AppData\Local\Temp\cpython-3.11.6+20231002-x86_64-pc-windows-msvc-shared-install_only.tar.gz to C:\Users\aarya\AppData\Local\Temp\hostpython3.11_859614a9
Compiling Python sources at C:\Users\aarya\AppData\Local\Temp\serious_python_tempb43f0909
Deleting Flutter bootstrap directory C:\Users\aarya\AppData\Local\Temp\flet_flutter_build_OYzPOnxyAd
```

**Additional information you deem important (e.g. issue happens only occasionally):**
I have NOT included flet, flet-core, and flet-runtime in requirements.txt as that led to another compilation error.

**Flet version (`pip show flet`)**:

```
(flet_venv) E:\flet_venv\jinkushal>pip show flet
Name: flet
Version: 0.18.0
Summary: Flet for Python - easily build interactive multi-platform apps in Python
Home-page:
Author: Appveyor Systems Inc.
Author-email: hello@flet.dev
License: Apache-2.0
Location: E:\flet_venv\Lib\site-packages
Requires: cookiecutter, flet-runtime, packaging, qrcode, watchdog, websocket-client, websockets
Required-by:
```

**Operating system**:
```
Edition	Windows 10 Home Single Language
Version	22H2
Installed on	‎24-‎11-‎2023
OS build	19045.3803
Experience	Windows Feature Experience Pack 1000.19053.1000.0
```

**Additional environment details:**
As the last step before error is `Compiling Python sources at C:\Users\aarya\AppData\Local\Temp\serious_python_tempb43f0909`
I feel the requirements.txt might help, so:

```
aenum==3.1.15
anyio==4.2.0
arrow==1.3.0
binaryornot==0.4.4
certifi==2023.7.22
chardet==5.2.0
charset-normalizer==3.3.2
click==8.1.7
colorama==0.4.6
cookiecutter==2.5.0
dnspython==2.4.2
h11==0.14.0
httpcore==0.17.3
httpx==0.24.1
idna==3.4
Jinja2==3.1.2
markdown-it-py==3.0.0
MarkupSafe==2.1.3
mdurl==0.1.2
oauthlib==3.2.2
packaging==23.1
Pygments==2.17.2
pymongo==4.6.1
pypng==0.20220715.0
python-dateutil==2.8.2
python-slugify==8.0.1
PyYAML==6.0.1
qrcode==7.4.2
repath==0.9.0
requests==2.31.0
rich==13.7.0
six==1.16.0
sniffio==1.3.0
text-unidecode==1.3
types-python-dateutil==2.8.19.14
typing_extensions==4.7.1
urllib3==2.1.0
watchdog==3.0.0
websocket-client==1.7.0
websockets==11.0.3
pytz==2023.3
```

**Comment by alan-copeland-keysight** [comment]
I am running into a very similar issue. Is there a way to get more debug errors out of the system?

**Comment by taaaf11** [comment]
Try adding `flet` into `requirements.txt` .

**Comment by frowNYfaceUnknown** [comment]
> Try adding `flet` into `requirements.txt` .

It still gives me the same error.

**Comment by taaaf11** [comment]
@frowNYfaceUnknown which packages are you using for the app?

**Comment by FeodorFitsner** [comment]
Try removing everything from `requirements.txt` except `flet`.

**Comment by alan-copeland-keysight** [comment]
As a debug test @FeodorFitsner? Or that is how its normally supposed to be structured?

**Comment by FeodorFitsner** [comment]
I mean where did you get the list of those requirements in the first place?

The reason for using just `flet` (plus, of course, other direct dependencies required by your app) is that it's replaced with `flet-embed` during the packaging which works on every platform.

But if you have dependencies like those (basically, recursive set of expanded `flet` deps) most of them are desktop only (like `watchdog`) and won't work on mobile.

**Comment by taaaf11** [comment]
@FeodorFitsner The solution is, only add top level dependencies. What do you think?

**Comment by frowNYfaceUnknown** [comment]
Yes, @FeodorFitsner thank you! I removed all the dependencies other than the direct dependencies (I had initially made the requirements.txt using `pip freeze > requirements.txt`) and the build now successfully generates the apk.

These are the packages used in the updated requirements.txt:
```
cookiecutter==2.5.0
dnspython==2.4.2
pymongo==4.6.1
pytz==2023.3
```
>What do you think?

Yes, for me, I only kept the libraries that were directly used in the code and the direct dependencies for them and it worked.
